core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1126,9 +1148,69 @@ public ServerBuilder tls(KeyManagerFactory keyManagerFactory) {
         return this;
     }
 
+    /**
+     * Sets the specified {@link TlsProvider} which will be used for building an {@link SslContext} of
+     * a hostname.
+     *
+     * <pre>{@code
+     * Server
+     *   .builder()
+     *   .tlsProvider(
+     *     TlsProvider.builder()
+     *                // Set the default key pair.
+     *                .setDefault(TlsKeyPair.of(...))
+     *                // Set the key pair for "example.com".
+     *                .set("example.com", TlsKeyPair.of(...))
+     *                .build())
+     * }</pre>
+     *
+     * <p>Note that this method mutually exclusive with {@link #tls(TlsKeyPair)} and other static TLS settings.
+     */
+    @UnstableApi
+    public ServerBuilder tlsProvider(TlsProvider tlsProvider) {
```

Comment:
It is possible to support `TlsProvider` for `VirtualHost`s. I didn't implement it because a `TlsProvider` can contain multiple certificates so one `TlsProvider` can cover all `VirtualHost`s. 

If there are any requirements for it, I'd be happy to implement it.